internal-messaging/message/cc/DataMessage.cc

```diff
@@ -3,96 +3,45 @@
 #include "DataMessage.h"
 #include <stdexcept>
 
-bool number_value(const char* string) {
```

Comment:
It's not immediately obvious what this function does. Can you add some comments?